File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        priority_queue<int> pq;
        for(int i=0;i<nums.size();i++)
        {
            for(int j=0;j<nums.size();++j)
            {
                if(i==j) continue;
                pq.push((nums[i]-1)*(nums[j]-1));
            }
        }
        return pq.top();
    }
};

File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.java
class Solution {
    public int maxProduct(int[] nums) {
        PriorityQueue<Integer> maxh=new PriorityQueue<>(Collections.reverseOrder());
        for(int curr:nums) 
            maxh.add(curr);
      
        return ((maxh.poll())-1)*((maxh.poll())-1);
    }
}
// PriorityQueue <Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        
//         for (int i = 0 ; i< nums.length; i++){
//             pq.add(nums[i]);
//         }
        
//         int first = (pq.poll() )-1;
//         int second = (pq.poll()) -1;
        
//         return first*second;